src/operators/CourseOperations.ts

```
@@ -0,0 +1,64 @@
+import Course, { CourseInterface } from '../models/course';
+import { v4 as uuidv4 } from 'uuid';
+import { checkUserId } from './index';
+
+const checkCourseUser = checkUserId(Course.get);
+
+export const create = async (course: CourseInterface) => {
+	try {
+		const hashKey = uuidv4();
+		course.id = hashKey;
+		await Course.create(course);
+		console.log(
+			'SUCCESS: Creation of course with id ' + course.id + ' successful.'
+		);
+	} catch (err) {
+		new Error('Failed to create course with id' + course.id);
+	}
+};
+
+export const read = async (key: string, userID: string) => {
+	try {
+		const course = await Course.get(key);
+
+		if (course && course.owner == userID) {
+			return course;
+		} else {
+			throw 'ERROR - could not read course with key ' + key;
+		}
+	} catch (err) {
+		console.error(err);
```

We should probably rethrow the error as well. `throw err;` at the end.